documentation/http_api_index.markdown

```diff
@@ -0,0 +1,141 @@
+---
+layout: default
+title: 'Puppet Server HTTP API: Index'
+---
+
+Puppet Server provides several services via HTTP API, and the Puppet agent application uses those services to resolve a node's credentials, retrieve a configuration catalog, retrieve file data, and submit reports.
+
+Many of these endpoints are the same as the [Puppet HTTP API](https://puppet.com/docs/puppet/latest/http_api/http_api_index.html).
+
+## V1/V2 HTTP APIs (removed)
+
+The V1 and V2 APIs were removed in Puppet 4.0.0. The routes that were previously under `/` or `/v2.0` can now be found under the [`/puppet/v3`](#puppet-v3-http-api) API or [`/puppet-ca/v1`](#ca-v1-http-api) API.
+
+Starting with Puppet Server 2.1, Server provides both the current and previous API endpoints, and can serve nodes running Puppet agent 3.x and 4.x. However, Rack masters, WEBrick masters, and Puppet Server 2.0 cannot serve nodes running Puppet 3.x.
+
+## Puppet and Puppet CA APIs
+
+Beginning with Puppet 4, Puppet's HTTP API was split into two separately versioned APIs:
+
+-   An API for configuration-related services
+-   An API for the certificate authority (CA).
+
+All configuration endpoints are prefixed with `/puppet`, while all CA endpoints are prefixed with `/puppet-ca`. All endpoints are explicitly versioned: the prefix is always immediately followed by a string like `/v3` (a directory separator, the letter `v`, and the version number of the API).
+
+### Authorization
+
+Authorization for `/puppet` endpoints is still controlled with Puppet's `auth.conf` authorization system.
```

Comment:
Its unclear if this is talking about old auth.conf or new auth.conf. The /puppet endpoints are managed by *both*, with (I think?) the general recommendation that old (ie Puppet's) auth.conf be set to `allow *` and new (ie Puppet Server's) auth.conf contain's the actual authorization rules. The `puppet-ca` routes are controlled solely by the new auth.conf.